### PR TITLE
TIP-1228: Adapter for PM indexer

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductModelProjection.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                    return $attribute->getCode();
+                }
+            )->toArray();
+
+        sort($attributesOfAncestors);
+
+        return $attributesOfAncestors;
+    }
+
+    private function getSortedAttributeCodes(ProductModelInterface $entityWithFamilyVariant): array
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductModelProjection.php]: weird naming for the parameter, but as it's a temporary class it does not matter

File: tests/back/Pim/Enrichment/Integration/ProductModel/Query/Sql/GetElasticsearchProductModelProjectionIntegration.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+class GetElasticsearchProductModelProjectionIntegration extends TestCase
+{
+    protected function getConfiguration(): Configuration
+    {
+        return $this->catalog->useTechnicalCatalog();
+    }
+
+    public function test_it_get_projection_of_sub_product_model()
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/ProductModel/Query/Sql/GetElasticsearchProductModelProjectionIntegration.php]:
```suggestion
    public function test_that_it_gets_the_projection_of_a_sub_product_model()
```

File: tests/back/Pim/Enrichment/Integration/ProductModel/Query/Sql/GetElasticsearchProductModelProjectionIntegration.php [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            'attributes_for_this_level' => [
+                'a_simple_select',
+                'a_text'
+            ]
+        ];
+
+        $this->checkProductModelProjectionFormat('child', $expected);
+    }
+
+    public function test_it_get_projection_of_root_product_model()
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/ProductModel/Query/Sql/GetElasticsearchProductModelProjectionIntegration.php]:
```suggestion
    public function test_that_it_gets_the_projection_of_a_root_product_model()
```

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductModelProjection.php (hunk 1 above)
Comment (pierallard) [reply src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/GetElasticsearchProductModelProjection.php]: Copy paste, and you're right, it does not matter. I will drop it.

**Pull request merged**